Analysis/src/TypeInfer.cpp

```diff
@@ -1803,6 +1803,8 @@ WithPredicate<TypeId> TypeChecker::checkExpr(const ScopePtr& scope, const AstExp
         result = checkExpr(scope, *a);
     else if (auto a = expr.as<AstExprIfElse>())
         result = checkExpr(scope, *a, expectedType);
+    else if (auto a = expr.as<AstExprInterpString>())
+        result = {stringType};
```

Comment:
Separately, if we allow `` `standalone` `` backtick string literals, you will want to apply the same logic we have for `AstExprConstantString` above, i.e.

```
local foo: "foo" = `foo`
```

We need the inference of the expression `` `foo` `` to be of type `"foo"`, not `string`, and we want to constrain this rule to only if the string interpolation expression has _no_ interpolating expressions.

There's other cases that's a little too complex to deal with, like the cross product of `` `foo {"bar" :: "bar" | "baz"}` `` is `"foo bar" | "foo baz"`, but this can't work in a greedy type system, so I wouldn't bother implementing this.

Reply:
Actually, with https://github.com/Roblox/luau/pull/614#discussion_r931653423 this is redundant, so the only case remaining are the complex ones.

Reply:
Should be redundant, but I want to write a test for this anyway.